core/platform/storage/cloud_storage_emulator.py

```diff
@@ -0,0 +1,188 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An emulator that mocks the core.platform.storage API."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals # pylint: disable=import-only-modules
+
+import pickle
+
+import feconf
+import python_utils
+
+import redis
+
+REDIS_CLIENT = redis.StrictRedis(
+    host=feconf.REDISHOST,
+    port=feconf.REDISPORT,
+    db=feconf.STORAGE_EMULATOR_REDIS_DB
+)
+
+
+class Blob(python_utils.OBJECT):
+    """Object for storing the file data."""
+
+    def __init__(self, name, data, content_type):
+        """Initialize blob.
+
+        Args:
+            name: str. The name of the blob.
+            data: str|bytes. The data of the blob. If the data are string,
```

Comment:
Data are always retrieved as bytes, I wanted to provide this support, since the library that we used in prod also supports this. I think that it would be nice to refactor our code not to rely on that fact, but I don't want to do it in this PR since there is still a lot work needed already.

Reply:
Should this be tracked in a follow-up? (with a todo here perhaps)

Reply:
Yes, done.